MiniAn/minian_main.py

```diff
@@ -0,0 +1,355 @@
+        # 1. Compute max projection
+        max_proj = save_minian(Y_fm_chk.max("frame").rename("max_proj"), **minian_parameters.params_save_minian).compute()
+        # 2. Generating over-complete set of seeds
+        with mn_utils.except_type_error("seeds_init"):
+            seeds = seeds_init(Y_fm_chk, **minian_parameters.params_seeds_init)
+
+        # 3. Peak-Noise-Ratio refine
+        print(mn_defs.Messages.INIT_SEEDS_PNR_REFI, flush=True)
+        with mn_utils.except_type_error("pnr_refine"):
+            seeds, pnr, gmm = pnr_refine(Y_hw_chk, seeds, **minian_parameters.params_pnr_refine)
+
+        # 4. Kolmogorov-Smirnov refine
+        print(mn_defs.Messages.INIT_SEEDS_KOLSM_REF, flush=True)
+        with mn_utils.except_type_error("ks_refine"):
+            seeds = ks_refine(Y_hw_chk, seeds, **minian_parameters.params_ks_refine)
+
+        # 5. Merge seeds
+        print(mn_defs.Messages.INIT_SEEDS_MERG, flush=True)
+        seeds_final = seeds[seeds["mask_ks"] & seeds["mask_pnr"]].reset_index(drop=True)
+        with mn_utils.except_type_error("seeds_merge"):
+            seeds_final = seeds_merge(Y_hw_chk, max_proj, seeds_final, **minian_parameters.params_seeds_merge)
+
+        return seeds_final, max_proj
+
+def component_initialization(Y_hw_chk, Y_fm_chk, seeds_final, intpath, chk, minian_parameters):
+    ### Component initialization ###
+    print(mn_defs.Messages.INIT_COMP, flush=True)
+    with mn_utils.except_print_error_no_cells(mn_defs.Messages.INIT_COMP):
+        # 1. Initialize spatial
+        print(mn_defs.Messages.INIT_COMP_SPATIAL, flush=True)
+        with mn_utils.except_type_error("initA"):
+            A_init = initA(Y_hw_chk, seeds_final[seeds_final["mask_mrg"]], **minian_parameters.params_initA)
+
+        A_init = save_minian(A_init.rename("A_init"), intpath, overwrite=True)
+        # 2. Initialize temporal
+        print(mn_defs.Messages.INIT_COMP_TEMP, flush=True)
+        C_init = initC(Y_fm_chk, A_init)
+        C_init = save_minian(C_init.rename("C_init"), intpath, overwrite=True,
+                            chunks={"unit_id": 1, "frame": -1})
+        # 3. Merge components
+        print(mn_defs.Messages.INIT_COMP_MERG, flush=True)
+        with mn_utils.except_type_error("unit_merge"):
+            A, C = unit_merge(A_init, C_init, **minian_parameters.params_unit_merge)
+
+        A = save_minian(A.rename("A"), intpath, overwrite=True)
+        C = save_minian(C.rename("C"), intpath, overwrite=True)
+        C_chk = save_minian(C.rename("C_chk"), intpath, overwrite=True,
+                            chunks={"unit_id": -1, "frame": chk["frame"]})
+        # 4. Initialize background
+        print(mn_defs.Messages.INIT_COMP_BACKG, flush=True)
+        b, f = update_background(Y_fm_chk, A, C_chk)
+        f = save_minian(f.rename("f"), intpath, overwrite=True)
+        b = save_minian(b.rename("b"), intpath, overwrite=True)
+
+    return A, C, C_chk, f, b
+
+
+def minian_main(minian_parameters):
```

Comment:
I'd keep the main functions on top, and all the rest of functions, in the order they're called

Reply:
I didn't know it would work. I was trying before and it wasn't working...
In fact this work
```
def 2add(a, b):
 return 2*add(a, b)

def add(a, b):
 return a + b
```

but apparently this don't work...
```
def 2add(a, b):
 return 2*add(a, b)

print(add(a, b))

def add(a, b):
 return a + b
```